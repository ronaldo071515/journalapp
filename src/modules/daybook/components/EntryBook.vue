<template>
    <div class="entry-container mb-3 pointer p-2"
        @click="$router.push({ name: 'entry', params: { id: 10 } })">
        <!-- titulo -->
        <div class="entry-title d-flex">
            <span class="text-success fs-5 fw-bold">15</span>
            <span class="mx-1 fs-4">julio</span>
            <span class="mx-2 fw-light">2023, jueves</span>
        </div>

        <div class="entry-description">
            Lorem ipsum dolor sit amet consectetur, adipisicing elit. Doloremque consequuntur ipsum, maiores in neque, quaerat odit error aliquam debitis voluptatum nihil harum id magni provident pariatur iste dolor labore! Ipsa!
        </div>
    </div>
</template>

<script>
export default {

}
</script>

<style lang="scss" scoped>
.entry-container {
    border-bottom: 1px solid #2c3e50;
    transition: 0.2s all ease-in;
    
    &:hover {
        background-color: lighten($color: grey, $amount: 45);
        transition: 0.2s all ease-in;
    }
    .entry-description {
        font-size: 12px;
    }
}


</style>