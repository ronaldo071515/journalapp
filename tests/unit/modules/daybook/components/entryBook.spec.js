import { shallowMount } from "@vue/test-utils";
import EntryBook from '@/modules/daybook/components/EntryBook.vue'
import { journalState } from '../../../mock-data/test-journal-state';

describe('Pruebas en el entry component', () => {

    const mockRouter = {
        push: jest.function
    }

    const wrapper = shallowMount( EntryBook, {
        props: {
            entry: journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    })

    test('debe de hacer match con el snapshot', () => {
        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('debe de redireccionar al hacer click en el entry-container', () => {
        
        const entryContainer = wrapper.find('.entry-container')
        entryContainer.trigger('click')

        expect( mockRouter.push ).toHaveBeenCalledWith({
            name: 'entry',
            params: {
                id: '-NNcu2p47VN3_Fvl5dIr'
            }
        })

    })

    test('pruebas en las propiedades computadas', () => {
        
    })

})