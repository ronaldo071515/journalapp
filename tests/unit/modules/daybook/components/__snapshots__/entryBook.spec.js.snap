// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el entry component debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- titulo -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">6</span><span class="mx-1 fs-4">Febrero</span><span class="mx-2 fw-light">2023, Lunes</span></div>
  <div class="entry-description">Hoy 06 de febrero de 2023, estoy aprendiendo vuejs y vuex con fernando herrera!</div>
</div>
`;
